x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportExplainLifecycleAction.java

```diff
@@ -165,9 +185,10 @@ static IndexLifecycleExplainResponse getIndexLifecycleExplainResponse(
 
         final IndexLifecycleExplainResponse indexResponse;
         if (metadata.isIndexManagedByILM(indexMetadata)) {
+            final IndexLifecycleMetadata indexLifecycleMetadata = metadata.custom(IndexLifecycleMetadata.TYPE);
+            final boolean policyExists = indexLifecycleMetadata.getPolicies().containsKey(policyName);
             // If this is requesting only errors, only include indices in the error step or which are using a nonexistent policy
-            if (onlyErrors == false
-                || (ErrorStep.NAME.equals(lifecycleState.step()) || indexLifecycleService.policyExists(policyName) == false)) {
+            if (onlyErrors == false || (ErrorStep.NAME.equals(lifecycleState.step()) || policyExists == false)) {
```

Comment:
The `IndexLifecycleService` only runs on the master node, which meant that `policyExists` was always `false`. I changed it to use the `Metadata` to check for policy existence.